data_structures/two_pointer.cpp

```diff
@@ -0,0 +1,77 @@
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+/**
+ * @brief Calculates the maximum water that can be trapped between vertical
+ * lines on a 2D plane.
+ *
```

Comment:
you could use this as your file documentation instead of the function documentation